### PR TITLE
Fix base rights for character devices

### PR DESCRIPTION
**Pull request by Pryz**
While testing the latest `python.wasm` from https://github.com/vmware-labs/webassembly-language-runtimes, we were not able to get the python shell working. Turns out this was a rights issue with `fd=0`.

Per https://github.com/WebAssembly/wasi-libc/blob/a6f871343313220b76009827ed0153586361c0d5/libc-bottom-half/sources/isatty.c#L13-L18, "A tty is a character device that we can't seek or tell on.".

This change re-use `fileRightsBase` and remove FD_SEEK and FD_TELL rights for all character devices.

### CODE REVIEW

**Review by achille-roussel — COMMENTED** [review]
Let's add a test in https://github.com/tetratelabs/wazero/blob/main/imports/wasi_snapshot_preview1/fs_test.go#L232 to guarantee that we won't introduce a regression.

**Comment by codefromthecrypt** [comment]
I can help add a test to this

**Review by codefromthecrypt — APPROVED** [review]
Thanks for getting to the bottom of this. A broke build is a bad day for everyone, so thanks for fixing it!

File: imports/wasi_snapshot_preview1/fs_test.go [hunk 1]
```diff
@@ -239,6 +240,16 @@ func Test_fdFdstatGet(t *testing.T) {
 	fsc := mod.(*wasm.ModuleInstance).Sys.FS()
 	preopen := fsc.RootFS()
 
+	// replace stdin with a fake TTY file.
+	// TODO: Make this easier once we have in-memory platform.File
```
Comment (codefromthecrypt) [inline imports/wasi_snapshot_preview1/fs_test.go:244]: This maybe by next week depending on how much zombification happens during intercontinental back to back travel

**Review by achille-roussel — APPROVED** [review]
🚢

**Comment by codefromthecrypt** [comment]
@achille-roussel can you take a look to see if this is legit in the go tests?

> 2023/05/12 00:26:03 write /tmp/go-example-stdout-Example_minimal-0.txt: Bad file number

**Comment by achille-roussel** [comment]
v1.1.0
```
$ PATH=$PATH:$PWD/misc/wasm GOROOT=$PWD GOARCH=wasm GOOS=wasip1 ./bin/go test ./src/archive/tar
ok  	archive/tar	1.132s
```

this PR
```
$ PATH=$PWD/../../github.com/stealthrocket/wazero:$PATH:$PWD/misc/wasm GOROOT=$PWD GOARCH=wasm GOOS=wasip1 ./bin/go test ./src/archive/tar
2023/05/12 00:46:00 write /var/folders/qc/zv1mt0j11sscgjygfx47nxlm0000gn/T//go-example-stdout-Example_minimal-0.txt: Bad file number
FAIL	archive/tar	1.151s
FAIL
```

main
```
$ PATH=$PWD/../../github.com/stealthrocket/wazero:$PATH:$PWD/misc/wasm GOROOT=$PWD GOARCH=wasm GOOS=wasip1 ./bin/go test ./src/archive/tar
2023/05/12 00:46:20 write /var/folders/qc/zv1mt0j11sscgjygfx47nxlm0000gn/T//go-example-stdout-Example_minimal-1.txt: Bad file number
FAIL	archive/tar	0.861s
FAIL
```

So we have a regression but it seems like it's not due to this change.

**Comment by codefromthecrypt** [comment]
thanks for the analysis @achille-roussel. I'll look into the broke main next!

**Pull request merged**